### ISSUE TITLE
Readonly relation custom-field causes bug when updating product variant on admin-UI

### ISSUE THREAD

**Issue by Cyp-Leg** [vendure-ecommerce/vendure#1403]
**Describe the bug**
When creating a new custom field, e.g a product variant custom field with a type 'relation' on another entity and make it readOnly, the Create/UpdateProductVariantCustomFieldsInput does not contain this field but when updating a product variant on Admin UI, the input contains this field, causing a GraphQL error

**To Reproduce**
1. Define a new relation as a custom field
2. Specify 'readOnly: true'
3. Go to the admin UI
4. Try to update any field of this entity (e.g product variant name) and click "Update"

**Expected behavior**
Product variant should be updated with only not-readonly fields in the request

**Environment (please complete the following information):**
 - @vendure/core version: 1.4.5
 - Nodejs version: 14.17
 - Database (mysql/postgres etc): postgres

**Comment by michaelbromley** [comment]
I've just been trying to reproduce this but can't so far. Here's what I did:

```ts
   customFields: {
        Customer: [
            {
                name: 'avatar',
                type: 'relation',
                entity: Asset,
                readonly: true,
            },
        ],
    },
```

Then in the Admin UI:

1. open Customer detail view
2. edit name field
3. click "update"
4. no error

This line https://github.com/vendure-ecommerce/vendure/blob/ecd1b1756d5fe94c228c5d76d1d44e3f4addcc3b/packages/admin-ui/src/lib/core/src/data/utils/remove-readonly-custom-fields.ts#L77 seems to be working to remove the `customFields.avatar` property from the input object as intended.

In my case, the value of `customFields.avatar` was `null`. Could you check the value in your case? Is it somehow `undefined`?

**Comment by michaelbromley** [comment]
From Slack, another user running into seeming the same issue:

---

f you add a readonly custom field, the admin UI renders the filed as disabled, but it also submits the value in the GraphQL mutation when you update the entity. This triggers a BAD_USER_INPUT message from the graphql backend. The message is not too helpful and there's no error logged, so kind of hard to debug, but when I removed the readonly flag in the field config, I stopped getting this error.

```ts
config.customFields.ProductVariant.push({
      type: 'int',
      name: 'totalSales',
      defaultValue: 0,
      readonly: true,
      label: [
        { languageCode: LanguageCode.en, value: 'Total Sales' },
      ],
    });
```

---
Note: also using postgres, in case that is the common factor?
